x-pack/plugins/case/server/common/models/commentable_case.ts

```diff
@@ -0,0 +1,300 @@
+      service: this.service,
+    });
+  }
+
+  /**
+   * Update a comment and update the corresponding case's update_at and updated_by fields.
+   */
+  public async updateComment({
+    updateRequest,
+    updatedAt,
+    user,
+  }: {
+    updateRequest: CommentPatchRequest;
+    updatedAt: string;
+    user: User;
+  }): Promise<UpdateCommentResp> {
+    const { id, version, ...queryRestAttributes } = updateRequest;
+
+    const [comment, commentableCase] = await Promise.all([
+      this.service.patchComment({
+        client: this.soClient,
+        commentId: id,
+        updatedAttributes: {
+          ...queryRestAttributes,
+          updated_at: updatedAt,
+          updated_by: user,
+        },
+        version,
+      }),
+      this.update({ date: updatedAt, user }),
+    ]);
+    return {
+      comment,
+      commentableCase,
+    };
+  }
+
+  /**
+   * Create a new comment on the appropriate case. This updates the case's updated_at and updated_by fields.
+   */
+  public async createComment({
+    createdDate,
+    user,
+    commentReq,
+  }: {
+    createdDate: string;
+    user: User;
+    commentReq: CommentRequest;
+  }): Promise<NewCommentResp> {
+    if (commentReq.type === CommentType.alert) {
+      if (this.status === CaseStatuses.closed) {
+        throw Boom.badRequest('Alert cannot be attached to a closed case');
+      }
+
+      if (!this.subCase && this.collection.attributes.type === CaseType.collection) {
+        throw Boom.badRequest('Alert cannot be attached to a collection case');
+      }
+    }
+
+    const [comment, commentableCase] = await Promise.all([
+      this.service.postNewComment({
+        client: this.soClient,
+        attributes: transformNewComment({
+          associationType: this.subCase ? AssociationType.subCase : AssociationType.case,
+          createdDate,
+          ...commentReq,
+          ...user,
+        }),
+        references: this.buildRefsToCase(),
+      }),
+      this.update({ date: createdDate, user }),
+    ]);
+    return {
+      comment,
+      commentableCase,
+    };
+  }
+
+  private formatCollectionForEncoding(totalComment: number) {
+    return {
+      id: this.collection.id,
+      version: this.collection.version ?? '0',
+      totalComment,
+      ...this.collection.attributes,
+      connector: transformESConnectorToCaseConnector(this.collection.attributes.connector),
+    };
+  }
+
+  public async encode(): Promise<CollectionWithSubCaseResponse> {
+    const collectionCommentStats = await this.service.getAllCaseComments({
+      client: this.soClient,
+      id: this.collection.id,
+      options: {
+        fields: [],
+        page: 1,
+        perPage: 1,
+      },
+    });
+
+    if (this.subCase) {
+      const subCaseComments = await this.service.getAllSubCaseComments({
+        client: this.soClient,
+        id: this.subCase.id,
+      });
+
+      return CollectWithSubCaseResponseRt.encode({
+        subCase: flattenSubCaseSavedObject({
```

Comment:
Shouldn't we return `subCases: [flattenSubCaseSavedObject(...)]`?

Reply:
Yeah I talked with Xavier about this and we thought for now we'd keep it as a single entry rather than an array of sub cases for the add comment response.